crates/agent-sql/src/discovers.rs

```diff
@@ -161,6 +163,32 @@ where
     Ok(())
 }
 
+/// Returns a map of catalog_name to md5 hash of the live spec. The map will only
+/// include entities that exist and have a non-null md5 hash.
+pub async fn fetch_spec_md5_hashes(
```

Comment:
future refactor / not blocking: seems like this could be pretty naturally folded into `build_merged_catalog` and it's delegate calls to `agent_sql::discovers::resolve_merge_target_specs`, rather than being a separate pass ?